src/service/job-summary/job-summary-service.ts

```diff
@@ -0,0 +1,150 @@
+      .addDetails("Post", post)
+      .addDetails("Local Execution", localExecution)
+      .write();
+  }
+
+  private constructPrePostResult(result: ExecuteCommandResult[]): string {
+    const prePostTableHeaders = [
+      { data: "Command", header: true },
+      { data: "Execution Result", header: true },
+      { data: "Execution Time", header: true },
+    ];
+    const data = result.map((res) => [res.command, this.getExecutionResultString(res.result), `${res.time}`]);
+    return core.summary
+      .emptyBuffer()
+      .addTable([prePostTableHeaders, ...data])
+      .stringify();
+  }
+
+  private constructExecutionResult(executionNodeResult: ExecuteNodeResult[], checkoutInfo: CheckedOutNode[]): string {
+    const tableHeaders = [
+      { data: "Project", header: true },
+      { data: "Source", header: true },
+      { data: "Target", header: true },
+      { data: "Merged", header: true },
+      { data: "Execution Result", header: true },
+      { data: "Avg Execution Time", header: true },
+    ];
+    return core.summary
+      .emptyBuffer()
+      .addTable([tableHeaders, ...this.getExecutionResultData(executionNodeResult, checkoutInfo)])
+      .addEOL()
+      .addRaw("```mermaid", true)
+      .addRaw(this.constructGraph(executionNodeResult), true)
+      .addRaw("```", true)
+      .stringify();
+  }
+
+  private getExecutionResult(executeCommandResults: ExecuteCommandResult[]): ExecutionResult {
+    return executeCommandResults.find((res) => res.result !== ExecutionResult.OK)?.result ?? ExecutionResult.OK;
+  }
+
+  private getExecutionResultString(result: ExecutionResult): string {
+    switch (result) {
+      case ExecutionResult.NOT_OK:
+        return "\u274C";
+      case ExecutionResult.SKIP:
+        return "&#9940;";
+      default:
+        return "\u2705";
+    }
+  }
+
+  private getExecutionResultData(executionResult: ExecuteNodeResult[], checkoutInfo: CheckedOutNode[]): string[][] {
+    return executionResult.map((res) => {
+      const nodeCheckoutInfo = checkoutInfo.find((info) => info.node.project === res.node.project)!.checkoutInfo;
+      const result = this.getExecutionResultString(this.getExecutionResult(res.executeCommandResults));
+
+      return [
+        res.node.project,
+        nodeCheckoutInfo ? `${nodeCheckoutInfo.targetGroup}/${nodeCheckoutInfo.targetName}:${nodeCheckoutInfo.targetBranch}` : "checkout skipped",
+        nodeCheckoutInfo ? `${nodeCheckoutInfo.sourceGroup}/${nodeCheckoutInfo.sourceName}:${nodeCheckoutInfo.sourceBranch}` : "checkout skipped",
+        nodeCheckoutInfo?.merge ? "\u2705" : "\u274C",
+        result,
+        res.executeCommandResults.length > 0
+          ? `${res.executeCommandResults.reduce((prev, curr) => prev + curr.time, 0) / res.executeCommandResults.length}`
+          : "0",
+      ];
+    });
+  }
+
+  private constructGraph(executionResult: ExecuteNodeResult[]) {
+    return `flowchart LR;
+        ${executionResult
+          .map((res) => {
```

Comment:
```suggestion
          .map(res => {
```

Reply:
Oof sorry about this. I use VS code formatter for quick formatting and i think it adds these parenthesis